benchmark/solver/solver.cpp

```diff
@@ -184,7 +184,9 @@ const std::map<std::string, std::function<std::unique_ptr<gko::LinOpFactory>(
         {"jacobi",
          [](std::shared_ptr<const gko::Executor> exec) {
              std::shared_ptr<const gko::LinOpFactory> f =
-                 gko::preconditioner::Jacobi<>::build().on(exec);
+                 gko::preconditioner::Jacobi<>::build()
+                     .with_skip_sorting(true)
```

Comment:
I don't entirely agree with this change - we don't skip sorting for the other preconditioners, and as far as I can tell, we don't sort the input prior to passing it to the preconditioner, so if the input `.mtx` file is sorted incorrectly, the preconditioner will silently fail.

Reply:
When reading a matrix from file, we always sort it by calling [`ensure_row_major_order()`](https://github.com/ginkgo-project/ginkgo/blob/develop/include/ginkgo/core/base/matrix_data.hpp#L463) inside [the `read()` function](https://github.com/ginkgo-project/ginkgo/blob/develop/core/base/mtx_io.cpp#L99). So whenever we read from a file, we can be sure  that we get a sorted input (which is why I was only able to reproduce the MFEM error inside MFEM and not when exporting it into a file, even though the file was unsorted).

Reply:
Ah, thanks for correcting me - I was looking for `sort` in the IO file. `ensure_row_major_order` feels a bit inconsistent in that regard, but that's a point for another time 😄